work-request-exporter/function_setup.tf

```diff
@@ -0,0 +1,60 @@
+resource "null_resource" "Login2OCIR" {
+  count = local.user_image_provided ? 0 : 1
+  provisioner "local-exec" {
+    command = "echo '${var.oci_docker_password}' |  docker login ${local.oci_docker_repository} --username ${local.ocir_namespace}/${var.oci_docker_username} --password-stdin"
+  }
+}
+
+### Repository in the Container Image Registry for the container images underpinning the function 
+resource "oci_artifacts_container_repository" "function_repo" {
+  # note: repository = store for all images versions of a specific container image - so it included the function name
+  depends_on     = [null_resource.Login2OCIR]
+  count          = local.user_image_provided ? 0 : 1
+  compartment_id = var.compartment_ocid
+  display_name   = "${local.ocir_repo_name}/${local.function_name}"
+  is_public      = false
+  defined_tags   = {}
+  freeform_tags  = local.freeform_tags
+
+  lifecycle {
+    ignore_changes = [
+      defined_tags["Oracle-Tags.CreatedBy"],
+      defined_tags["Oracle-Tags.CreatedOn"],
+    ]
+  }
+}
+
+# ### build the function into a container image and push that image to the repository in the OCI Container Image Registry
+resource "null_resource" "FnImagePushToOCIR" {
```

Comment:
Is there really no way to use an existing image and build/push it ourselves outside of terraform? This all seems very odd to me

Reply:
But I guess the dockerfile and python function would need to be somewhere outside of platform-configs? I think we can definitely improve this if we have time